internal/ethapi/api.go

```diff
@@ -626,12 +627,19 @@ func (api *BlockChainAPI) ChainId() *hexutil.Big {
 }
 
 // GetActivePrecompilesAt returns the active precompile configs at the given block timestamp.
-func (s *BlockChainAPI) GetActivePrecompilesAt(ctx context.Context, blockTimestamp *big.Int) params.PrecompileUpgrade {
+func (s *BlockChainAPI) GetActivePrecompilesAt(ctx context.Context, blockTimestamp *big.Int) params.ChainConfigPrecompiles {
 	if blockTimestamp == nil {
 		blockTimestampInt := s.b.CurrentHeader().Time
 		blockTimestamp = new(big.Int).SetUint64(blockTimestampInt)
 	}
-	return s.b.ChainConfig().GetActivePrecompileUpgrade(blockTimestamp)
+	res := make(params.ChainConfigPrecompiles)
+	for _, module := range modules.RegisteredModules() {
+		if config := s.b.ChainConfig().GetActivePrecompileConfig(module.Address, blockTimestamp); config != nil && !config.IsDisabled() {
+			res[module.ConfigKey] = config
+		}
+	}
+
+	return res
```

Comment:
Can we move this back to being a function on `*params.ChainConfig` instead of requiring this logic here for marshaling?

Reply:
moved